dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java

```diff
@@ -701,6 +714,62 @@ public void testGetSchema_GivenEELicense_EnterpriseBaseTypeCollectionsShouldBeAv
                 +"BaseTypeCollection"));
     }
 
+    /**
+     * This method tests that given a {@link Field} of type {@link com.dotcms.contenttype.model.field.FileField}
+     * or {@link com.dotcms.contenttype.model.field.ImageField}, the following GraphQL fields are
+     * available to query:
+     *
+     * {@link FileAssetContentType#FILEASSET_FILE_NAME_FIELD_VAR}
+     * {@link FileAssetContentType#FILEASSET_DESCRIPTION_FIELD_VAR}
+     * {@link FileAssetContentType#FILEASSET_FILEASSET_FIELD_VAR}
+     * {@link FileAssetContentType#FILEASSET_METADATA_FIELD_VAR}
+     * {@link FileAssetContentType#FILEASSET_SHOW_ON_MENU_FIELD_VAR}
+     * {@link FileAssetContentType#FILEASSET_SORT_ORDER_FIELD_VAR}
+     */
+
+    @Test
+    public void testAvailableGraphQLFieldsOnImageAndFileFields()
+            throws DotDataException, DotSecurityException {
+        ContentType contentType = null;
+        try {
+            contentType = new ContentTypeDataGen().nextPersisted();
+            final Field fileField = new FieldDataGen().contentTypeId(contentType.id())
+                    .type(FileField.class).nextPersisted();
+            final Field imageField = new FieldDataGen().contentTypeId(contentType.id())
+                    .type(ImageField.class).nextPersisted();
+
+            APILocator.getGraphqlAPI().invalidateSchema();
+
+            final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
+
+            final GraphQLFieldDefinition fileFieldDefinition = schema
+                    .getObjectType(contentType.variable())
+                    .getFieldDefinition(fileField.variable());
+
+            final GraphQLFieldDefinition imageFieldDefinition = schema
+                    .getObjectType(contentType.variable())
+                    .getFieldDefinition(imageField.variable());
+
+            assertEquals(CustomFieldType.FILEASSET.getType(), fileFieldDefinition.getType());
+
+            assertTrue(areFileassetFieldsPresent((GraphQLObjectType) fileFieldDefinition.getType()));
+            assertTrue(areFileassetFieldsPresent((GraphQLObjectType) imageFieldDefinition.getType()));
+
+            assertEquals(CustomFieldType.FILEASSET.getType(), imageFieldDefinition.getType());
+        } finally {
+            APILocator.getContentTypeAPI(APILocator.systemUser()).delete(contentType);
+        }
+    }
+
+    private boolean areFileassetFieldsPresent(final GraphQLObjectType objectType) {
+        List<String> fileAssetFields = list(FILEASSET_FILE_NAME_FIELD_VAR,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'fileAssetFields' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028723)